clean/lake_charles_pd_cprr.py

```diff
@@ -120,30 +192,330 @@ def assign_agency(df):
+        .str.replace(r'(mg kaughenbaugh|c\. daughen baugh)',
+                     'marie daughenbaugh', regex=True)\
+        .str.replace(r'j. eving', 'j. ewing/marie daughenbaugh', regex=True)\
+        .str.replace(r'^2. harrell', 'r. harrell', regex=True)\
+        .str.replace(r'i?\.? ?stickell', 'j. stickell', regex=True)\
+        .str.replace(r'lig janice', 'robert janice', regex=True)\
+        .str.replace(r'^p\.$', '', regex=True)\
+        .str.replace(r'(\w{1})\.? holiday', r'k. holiday', regex=True)\
+        .str.replace(r"s\.? m ?'?[dbp]aniel?/?", 's. mcdaniel', regex=True)\
+        .str.replace(r'c\. booth', 'chad booth', regex=True)\
+        .str.replace(r'(\w+) /(\w+)', r' \1/\2', regex=True)\
+        .str.replace(r'^j. antonistration$', 'j. redd/n. stratton', regex=True)\
+        .str.replace(r't\.? bup[fl]echan', 't. suplechan', regex=True)\
+        .str.replace(' flurcy/p.', 'michael flurry/c. breland', regex=True)\
+        .str.replace(r'b?\.? ?randolph', 'benjamin randolph', regex=True)\
+        .str.replace(' comville', ' courville', regex=False)\
+        .str.replace(' veiller', ' veillon', regex=False)\
+        .str.replace(r'c\.  ?manu?el?', 'carlos manuel', regex=True)\
+        .str.replace(r'^a.  ake/c.boudin$', 'a. ake/c. boudin', regex=True)\
+        .str.replace('cbbeland', 'c. breland', regex=False)\
+        .str.replace('hammee', 'hammer', regex=False)\
+        .str.replace(' manual', ' manuel', regex=False)\
+        .str.replace(r'^ryjjennis$', 'russell dennis', regex=True)\
+        .str.replace(' \bdenyis\b', ' dennis', regex=True)\
+        .str.replace(' williays', ' williams', regex=False)\
+        .str.replace(r'(\w+) \b(\w{1})\b', r'\2 \1', regex=True)\
+        .str.replace(r'\.', '', regex=True)\
+        .str.replace(r'^(\w+)$', r' \1', regex=True)\
+        .str.replace(r'^ i$', 'jonathan landrum', regex=True)\
+        .str.replace(r'^j n', 'j', regex=True)
+    return df
+
+
+def assign_missing_names_19(df):
+    df.loc[(df.tracking_number == '16-7'), 'officer_s_accused'] = 'a malveaux'
+    df.loc[(df.tracking_number == '19-54'), 'officer_s_accused'] = 'a aeheb'
+    df.loc[(df.tracking_number == '17-36'), 'officer_s_accused'] = 'john saunier'
+    df.loc[(df.tracking_number == '17-38'), 'officer_s_accused'] = 'j littleton'
+    return df
+
+
+def split_rows_with_multiple_officers_19(df):
+    i = 0
+    for idx in df[df.officer_s_accused.str.contains("/")].index:
+        s = df.loc[idx + i, "officer_s_accused"]
+        parts = re.split(r"\s*(?:\/)\s*", s)
+        df = duplicate_row(df, idx + i, len(parts))
+        for j, name in enumerate(parts):
+            df.loc[idx + i + j, "officer_s_accused"] = name
+        i += len(parts) - 1
+    return df
+
+
+def review_first_names_from_post_19(df):
+    df = pd.read_csv(data_file_path('match/lake_charles_pd_cprr_2014_2019_assign_first_name_from_post_pprr_2020_11_06.xlsx'))
```

Comment:
Same as above.

Reply:
Makes much more sense to review in notebook. Thanks